java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -479,6 +448,45 @@ void executeInstruction(Instruction instruction) {
     checkerDispatcher.executeCheckPostStatement(instruction);
   }
 
+  private boolean handleMethodInvocation(Instruction instruction) {
+    boolean isStatic = instruction.opcode == Opcodes.INVOKESTATIC;
+    int arity = isStatic ? instruction.arity() : (instruction.arity() + 1);
+    ProgramState.Pop pop = programState.unstackValue(arity);
+    Preconditions.checkState(pop.values.size() == arity, "Arguments mismatch for INVOKE");
+    // TODO use constraintManager.createMethodSymbolicValue to create relational SV for equals
+    SymbolicValue returnSV = constraintManager.createSymbolicValue(instruction);
+    if (isStatic) {
+      // follow only static invocations for now.
+      String signature = instruction.fieldOrMethod.completeSignature();
+      MethodBehavior methodInvokedBehavior = behaviorCache.get(signature);
+      if (methodInvokedBehavior != null && methodInvokedBehavior.isComplete()) {
+        methodInvokedBehavior
+          .happyPathYields()
+          .forEach(yield ->
+            yield.statesAfterInvocation(Lists.reverse(pop.values), Collections.emptyList(), pop.state, () -> returnSV).forEach(ps -> {
+              checkerDispatcher.methodYield = yield;
+              checkerDispatcher.addTransition(ps);
+              checkerDispatcher.methodYield = null;
+            }));
+        methodInvokedBehavior
+          .exceptionalPathYields()
+          .forEach(yield ->
+            yield.statesAfterInvocation(
+              Lists.reverse(pop.values), Collections.emptyList(), pop.state, () -> constraintManager.createExceptionalSymbolicValue(yield.exceptionType())).forEach(ps -> {
+              ps.storeExitValue();
+              enqueue(new ProgramPoint(exitBlock), ps);
+            }));
+        return true;
+      }
+    }
+    if (instruction.hasReturnValue()) {
```

Comment:
matter of taste and style but I tend to prefer : 
```
programState = pop.state;
if(!instruction.hasReturnValue()) {
   programState = programeState.stackValue(returnSV); 
}
```
one less branch...

Reply:
and the naming of the method was actually inversed